src/pudl/extract/ferc1.py

```diff
@@ -897,7 +916,76 @@ def extract_dbf_generic(
             "min_year": min(ferc1_settings.dbf_years),
             "max_year": max(ferc1_settings.dbf_years),
         },
-    )
+    ).assign(sched_table_name=table_name)
+
+
+def extract_xbrl_generic(
+    table_names: list[str],
+    period: Literal["duration", "instant"],
+    ferc1_engine: sa.engine.Engine,
+    ferc1_settings: Ferc1Settings,
+) -> pd.DataFrame:
+    """Combine multiple raw xbrl instant or duration tables into one.
+
+    Args:
+        table_names: The list of raw table names provided in TABLE_NAME_MAP_FERC1
+            under xbrl. These are the tables you want to combine.
+        period: Either duration or instant, specific to xbrl data.
+        ferc1_engine: An SQL Alchemy connection engine for the FERC Form 1 database.
+        ferc1_settings: Object containing validated settings relevant to FERC Form 1.
+            Contains the tables and years to be loaded into PUDL.
+
+    There are some instances where multiple xbrl tables ought to be combined into
+    a single table to best mesh with data from the other source. This function
+    concatinates those tables into one. It is similar to the extract_dbf_generic
```

Comment:
nit: `concatenate`. because you might want to to do it to TEN tables

Reply:
haha thanks for catching this!

Reply:
There were two instances of this :eyes: